src/firmware/app/control/trajectory_planner.h

```diff
@@ -56,6 +35,49 @@ typedef struct FirmwareRobotPathParameters
     float final_speed;
 } FirmwareRobotPathParameters_t;
 
+// Struct that defines a single point on a trajectory
+// Includes the Position and Time data corresponding to that point
+typedef struct PositionTrajectoryElement
+{
+    Vector2d_t position;
+    float time;
+} PositionTrajectoryElement_t;
+
+typedef struct PositionTrajectory
+{
+    PositionTrajectoryElement_t* trajectory_elements;
+    FirmwareRobotPathParameters_t path_parameters;
+    float* speed_profile;
+} PositionTrajectory_t;
+
+// Struct that defines a single point on a velocity trajectory
+// Includes the velocity and Time data corresponding to that point
+typedef struct VelocityTrajectoryElement
+{
+    Vector2d_t linear_velocity;
+    float angular_velocity;
+    float time;
+} VelocityTrajectoryElement_t;
+
+typedef struct VelocityTrajectory
+{
+    VelocityTrajectoryElement_t* trajectory_elements;
+    FirmwareRobotPathParameters_t path_parameters;
+} VelocityTrajectory_t;
+
+typedef enum TrajectoryPlannerGenerationStatus
+{
+    OK                        = 0,
+    FINAL_VELOCITY_TOO_HIGH   = 1,
+    INITIAL_VELOCITY_TOO_HIGH = 2,
+
+    // INTERPOLATION_ELEMENT_MAXED_OUT is returned when the constant time interpolation
+    // function uses up all the avaiable array space provided to it. This can happen
+    // because constant interpolation period trajectories do not necessarily have the same
+    // number of elements as their constant arc-length counterparts
+    INTERPOLATION_ELEMENT_MAXED_OUT = 3,
```

Comment:
If we don't need numbers here I don't think we should put them (ie. just allow them to be auto-numbered), as it's less confusing and prevents two having the same value by accident.

Reply:
Removed the numbering 👍